lib/kubernetes-deploy/restart_task.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+require 'kubernetes-deploy/kubeclient_builder'
+
+module KubernetesDeploy
+  class RestartTask
+    include KubernetesDeploy::KubeclientBuilder
+
+    HTTP_OK_RANGE = 200..299
+
+    def initialize(context:, namespace:, logger:)
+      @context = context
+      @namespace = namespace
+      @logger = logger
+      @v1beta1_kubeclient = build_v1beta1_kubeclient(context)
+    end
+
+    def perform(deployments)
+      if deployments.empty?
+        raise ArgumentError, "#perform takes at least one deployment to restart"
+      end
+
+      fetch_deployments(deployments).each do |record|
```

Comment:
Does this mean each deployment is restarted in serial? How long does that take? Won't this only give the status that it's _starting_ to restart, not that it's actually successfully restarted all, say, 10,000 pods?